Firestore/core/test/firebase/firestore/util/filesystem_test.cc

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/util/filesystem.h"
+
+#if defined(_WIN32)
+#include <cwchar>
+#endif
+#include <fstream>
+
+#include "Firestore/core/src/firebase/firestore/util/autoid.h"
+#include "Firestore/core/src/firebase/firestore/util/log.h"
+#include "Firestore/core/src/firebase/firestore/util/path.h"
+#include "Firestore/core/test/firebase/firestore/util/status_test_util.h"
+#include "absl/strings/match.h"
+#include "absl/types/optional.h"
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace util {
+
+/** Creates an empty file at the given path. */
+static void Touch(const Path& path) {
+  std::ofstream out{path.native_value()};
```

Comment:
Visual C++ contains additional overloads to support filenames specified as wchar_t arrays. See, e.g. 

https://docs.microsoft.com/en-us/cpp/standard-library/basic-ofstream-class#open

As far as I can tell this is nonstandard, but works fine. Note that it does work in Windows: https://github.com/firebase/firebase-ios-sdk/pull/1634/commits/f28d8abcda858b9feff5bec8cf233309ceab8ad4 runs this this same test on all the platforms.

Reply:
Hmm, yes, but if I remember correctly, `native_value()` returns `std::wstring` which doesn't convert to a `wchar` pointer implicitly, so those overloads shouldn't work. Basically, I'm just slightly concerned that this might fail on older versions of Visual Studio. OTOH, it's no big deal to leave as is and fix later if necessary.

Reply:
Let's wait and see.